App crashed when network is unstable

**App version:** 1.7.1 based custom build from revision https://github.com/AntennaPod/AntennaPod/commit/0a9735e7

**Android version**: 8.1

**Device model**: Nexus 6 (Emulator)

**Expected behaviour**: App does not crash

**Current behaviour**: App crashed after press back

**Steps to reproduce**:

1. In the `Add Podcast` page, click `SEARCH ITUNES`
2. Before any responds received, downgrade the network speed to 2G
3. Wait for a while, and recover the network speed before any responds received
4. Click `navigate up` (i.e. `<-` on top-left) button
5. The app then crashed

**Stacktrace/Logcat**: 

No stacktrace found in logcat.

Stacktrace:

```
io.reactivex.exceptions.UndeliverableException: 
  at io.reactivex.plugins.RxJavaPlugins.onError (RxJavaPlugins.java:367)
  at io.reactivex.internal.operators.single.SingleCreate$Emitter.onError (SingleCreate.java:81)
  at de.danoeh.antennapod.fragment.ItunesSearchFragment.lambda$null$0 (ItunesSearchFragment.java:149)
  at de.danoeh.antennapod.fragment.-$$Lambda$ItunesSearchFragment$8SKiker3uTBCeDwaX4bzXtWDgUc.subscribe (Unknown Source:4)
  at io.reactivex.internal.operators.single.SingleCreate.subscribeActual (SingleCreate.java:39)
  at io.reactivex.Single.subscribe (Single.java:3438)
  at io.reactivex.internal.operators.single.SingleSubscribeOn$SubscribeOnObserver.run (SingleSubscribeOn.java:89)
  at io.reactivex.Scheduler$DisposeTask.run (Scheduler.java:578)
  at io.reactivex.internal.schedulers.ScheduledRunnable.run (ScheduledRunnable.java:66)
  at io.reactivex.internal.schedulers.ScheduledRunnable.call (ScheduledRunnable.java:57)
  at java.util.concurrent.FutureTask.run (FutureTask.java:266)
  at java.util.concurrent.ScheduledThreadPoolExecutor$ScheduledFutureTask.run (ScheduledThreadPoolExecutor.java:301)
  at java.util.concurrent.ThreadPoolExecutor.runWorker (ThreadPoolExecutor.java:1167)
  at java.util.concurrent.ThreadPoolExecutor$Worker.run (ThreadPoolExecutor.java:641)
  at java.lang.Thread.run (Thread.java:764)
Caused by: java.io.InterruptedIOException: 
  at okhttp3.internal.http2.Http2Stream.waitForIo (Http2Stream.java:579)
  at okhttp3.internal.http2.Http2Stream.takeResponseHeaders (Http2Stream.java:143)
  at okhttp3.internal.http2.Http2Codec.readResponseHeaders (Http2Codec.java:125)
  at okhttp3.internal.http.CallServerInterceptor.intercept (CallServerInterceptor.java:88)
  at okhttp3.internal.http.RealInterceptorChain.proceed (RealInterceptorChain.java:147)
  at okhttp3.internal.http.RealInterceptorChain.proceed (RealInterceptorChain.java:121)
  at de.danoeh.antennapod.core.service.download.AntennapodHttpClient.lambda$newBuilder$0 (AntennapodHttpClient.java:88)
  at de.danoeh.antennapod.core.service.download.-$$Lambda$AntennapodHttpClient$2Xq9YggjpSacmHedsJk0juFgtBc.intercept (Unknown Source)
  at okhttp3.internal.http.RealInterceptorChain.proceed (RealInterceptorChain.java:147)
  at okhttp3.internal.connection.ConnectInterceptor.intercept (ConnectInterceptor.java:45)
  at okhttp3.internal.http.RealInterceptorChain.proceed (RealInterceptorChain.java:147)
  at okhttp3.internal.http.RealInterceptorChain.proceed (RealInterceptorChain.java:121)
  at okhttp3.internal.cache.CacheInterceptor.intercept (CacheInterceptor.java:93)
  at okhttp3.internal.http.RealInterceptorChain.proceed (RealInterceptorChain.java:147)
  at okhttp3.internal.http.RealInterceptorChain.proceed (RealInterceptorChain.java:121)
  at okhttp3.internal.http.BridgeInterceptor.intercept (BridgeInterceptor.java:93)
  at okhttp3.internal.http.RealInterceptorChain.proceed (RealInterceptorChain.java:147)
  at okhttp3.internal.http.RetryAndFollowUpInterceptor.intercept (RetryAndFollowUpInterceptor.java:126)
  at okhttp3.internal.http.RealInterceptorChain.proceed (RealInterceptorChain.java:147)
  at okhttp3.internal.http.RealInterceptorChain.proceed (RealInterceptorChain.java:121)
  at okhttp3.RealCall.getResponseWithInterceptorChain (RealCall.java:200)
  at okhttp3.RealCall.execute (RealCall.java:77)
  at de.danoeh.antennapod.fragment.ItunesSearchFragment.lambda$null$0 (ItunesSearchFragment.java:137)
```

--- a/app/src/main/java/de/danoeh/antennapod/fragment/ItunesSearchFragment.java
+++ b/app/src/main/java/de/danoeh/antennapod/fragment/ItunesSearchFragment.java
@@ -5,6 +5,7 @@ import android.os.Bundle;
 import android.support.v4.app.Fragment;
 import android.support.v4.view.MenuItemCompat;
 import android.support.v7.widget.SearchView;
+import android.support.annotation.NonNull;
 import android.util.Log;
 import android.view.LayoutInflater;
 import android.view.Menu;
@@ -106,7 +107,7 @@ public class ItunesSearchFragment extends Fragment {
     }
 
     @Override
-    public View onCreateView(LayoutInflater inflater, ViewGroup container,
+    public View onCreateView(@NonNull LayoutInflater inflater, ViewGroup container,
                              Bundle savedInstanceState) {
         // Inflate the layout for this fragment
         View root = inflater.inflate(R.layout.fragment_itunes_search, container, false);
@@ -146,7 +147,9 @@ public class ItunesSearchFragment extends Fragment {
                                     emitter.onError(new IOException(prefix + response));
                                 }
                             } catch (IOException | JSONException e) {
-                                emitter.onError(e);
+                                if (!disposable.isDisposed()) {
+                                    emitter.onError(e);
+                                }
                             }
                         })
                         .subscribeOn(Schedulers.io())
@@ -249,7 +252,9 @@ public class ItunesSearchFragment extends Fragment {
                         List<Podcast> podcasts = parseFeed(feedString);
                         emitter.onSuccess(podcasts);
                     } catch (IOException | JSONException e) {
-                        emitter.onError(e);
+                        if (!disposable.isDisposed()) {
+                            emitter.onError(e);
+                        }
                     }
                 })
                 .subscribeOn(Schedulers.io())

